File: 283-Move-Zeroes/283-Move-Zeroes.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        """
        Do not return anything, modify nums in-place instead.
        """
        # back = len(nums) - 1
        # for i in range(len(nums)):
        #     if nums[i] == 0 and i < back :
        #         nums[i],nums[back] = nums[back],nums[i]
        #         back -= 1
        count = 0
        for i in nums:
            if i == 0:
                count += 1
        for i in range(count):
            nums.remove(0)
        for i in range(count):
            nums.append(0)
            

File: 283-Move-Zeroes/283-Move-Zeroes.java
class Solution {
    public void moveZeroes(int[] nums) {
        if(nums.length != 1){
            int left = 0;
            int right = 0;
            while(left < nums.length && nums[left] != 0 ){
                left++;
            }
            if(left < nums.length - 1){

            right = left + 1;
            while(right < nums.length){
                if(nums[right] != 0){
                    int temp = nums[left];
                    nums[left] = nums[right];
                    nums[right] = temp;
                    left++;
                    right++;
                }
                else {
                    right++;
                }
            }
            }

        }
    }
}